pandas/io/parquet.py

```diff
@@ -108,18 +112,16 @@ def write(
                 **kwargs,
             )
         else:
-            self.api.parquet.write_table(table, path, compression=compression, **kwargs)
+            self.api.parquet.write_table(
+                table, file_obj, compression=compression, **kwargs
+            )
 
     def read(self, path, columns=None, **kwargs):
-        path, _, _, should_close = get_filepath_or_buffer(path)
-
-        kwargs["use_pandas_metadata"] = True
-        result = self.api.parquet.read_table(
-            path, columns=columns, **kwargs
-        ).to_pandas()
-        if should_close:
-            path.close()
-
+        parquet_ds = self.api.parquet.ParquetDataset(
+            path, filesystem=get_fs_for_path(path), **kwargs
```

Comment:
Is this `filesystem=get_fs_for_path(path)` needed? What happens if you just pass the path? (which I assume has eg a `s3://..` in it?)

Reply:
pyarrow seems to only allow a file path opposed to a dir path. Removing filesystem arg here throws:
```
            for path in path_or_paths:
                if not fs.isfile(path):
                    raise IOError('Passed non-file path: {0}'
>                                 .format(path))
E                   OSError: Passed non-file path: s3://pandas-test/parquet_dir

../../../.conda/envs/pandas-dev/lib/python3.7/site-packages/pyarrow/parquet.py:1229: OSError
```
To repo see the test case `test_s3_roundtrip_for_dir` I wrote below

Reply:
Ah, OK. I see now in pyarrow that apparently string URIs with "s3://..." are not supported (while "hdfs://" is supported). That's something we should fix on the pyarrow side as well. But of course until then this is fine.